pages/admin/index.tsx

```diff
@@ -104,44 +117,82 @@ const COLUMNS = [
-    requestType: 'Replacement',
-    status: 'COMPLETED',
-  },
-];
+// Application data for table
+type ApplicationData = {
+  name: {
+    name: string;
+    rcdUserId: number | undefined;
+  };
+  dateReceived: Date;
+  permitType: string;
+  requestType: string;
+  status: ApplicationStatus | undefined;
+};
+
+// Max number of entries in a page
+const PAGE_SIZE = 20;
 
 // Internal home page - view APP requests
 export default function Requests() {
-  const [permitTypeFilter, setPermitTypeFilter] = useControllableState({ defaultValue: 'All' });
-  const [requestTypeFilter, setRequestTypeFilter] = useControllableState({ defaultValue: 'All' });
+  //Filters
+  const [statusFilter, setStatusFilter] = useState<ApplicationStatus>();
+  const [permitTypeFilter, setPermitTypeFilter] = useState<PermitType>();
+  const [requestTypeFilter, setRequestTypeFilter] = useState<string>();
+  const [searchFilter, setSearchFilter] = useState<string>('');
+
+  // Sorting
+  const [sortOrder, setSortOrder] = useState<SortOptions>([['dateReceived', SortOrder.DESC]]);
+
+  // Debounce search filter so that it only gives us latest value if searchFilter has not been updated within last 500ms.
+  // This will avoid firing a query for each key the user presses
+  const debouncedSearchFilter = useDebounce<string>(searchFilter, 500);
+
+  // Data & pagination
+  const [requestsData, setRequestsData] = useState<ApplicationData[]>();
+  const [pageNumber, setPageNumber] = useState(0);
+  const [recordsCount, setRecordsCount] = useState(0);
+
+  // Make query to applications resolver
+  useQuery<GetApplicationsResponse, GetApplicationsRequest>(GET_APPLICATIONS_QUERY, {
+    variables: {
+      filter: {
+        order: sortOrder,
+        permitType: permitTypeFilter,
+        requestType: requestTypeFilter,
+        status: statusFilter,
+        search: debouncedSearchFilter,
+        offset: pageNumber * PAGE_SIZE,
+        limit: PAGE_SIZE,
+      },
+    },
+    notifyOnNetworkStatusChange: true,
+    onCompleted: data => {
+      setRequestsData(
+        data.applications.result.map(record => ({
+          name: {
+            name: record.firstName + ' ' + record.lastName,
+            rcdUserId: record.applicantId || undefined,
+          },
+          dateReceived: record.createdAt,
+          permitType: permitTypeString[record.permitType],
+          requestType: record.isRenewal ? 'Renewal' : 'Replacement',
+          status: record.applicationProcessing?.status || undefined,
+        }))
+      );
+      setRecordsCount(data.applications.totalCount);
+    },
+  });
+
+  // Map uppercase enum strings to lowercase
+  const permitTypeString: Record<PermitType, string> = {
+    [PermitType.Permanent]: 'Permanent',
+    [PermitType.Temporary]: 'Temporary',
+  };
+
+  // Set page number to 0 after every filter or sort change
+  useEffect(() => {
+    setPageNumber(0);
+  }, [statusFilter, permitTypeFilter, requestTypeFilter, debouncedSearchFilter, sortOrder]);
```

Comment:
Do we need sortOrder here?

Reply:
Yeah we do assuming that when the sorting order changes we want to go back to the first page (which I assume is the case because the data on the current page will most likely be different).

Reply:
Agreed